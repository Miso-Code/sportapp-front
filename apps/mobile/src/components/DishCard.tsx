import React from 'react'
import { StyleSheet, View } from 'react-native'
import { Card, Text, Chip, useTheme, MD3Theme } from 'react-native-paper'
import { useTranslation } from 'react-i18next'

interface DishCardProps {
	title: string
	weekday: string
	category: string
	calories: number
	protein: number
	carbs: number
	fats: number
}

const DishCard: React.FC<DishCardProps> = ({
	title,
	weekday,
	category,
	calories,
	protein,
	carbs,
	fats
}) => {
	const theme = useTheme()
	const styles = createStyles(theme)
	const { t } = useTranslation()

	return (
		<Card elevation={1} style={styles.card}>
			<Card.Content>
				<Text variant='labelLarge' style={styles.label}>
					{t(`nutritionalPlan.weekday.${weekday}`)}
				</Text>
				<Text variant='titleLarge' style={styles.cardTitle}>
					{title}
				</Text>
				<View style={styles.wrap}>
					<Chip compact disabled mode='outlined'>
						{calories} {t(`nutritionalPlan.macros.calories`)}
					</Chip>
					<Chip compact disabled mode='outlined'>
						{carbs}g {t(`nutritionalPlan.macros.of_carbs`)}
					</Chip>
					<Chip compact disabled mode='outlined'>
						{protein}g {t(`nutritionalPlan.macros.of_proteins`)}
					</Chip>
					<Chip compact disabled mode='outlined'>
						{fats}g {t(`nutritionalPlan.macros.of_fats`)}
					</Chip>
				</View>
				<Chip
					testID='chip'
					style={styles.chipToday}
					textStyle={styles.chipTodayText}
					compact>
					{t(`nutritionalPlan.dishType.${category}`)}
				</Chip>
			</Card.Content>
		</Card>
	)
}

const createStyles = (theme: MD3Theme) =>
	StyleSheet.create({
		card: {
			backgroundColor: '#f2f2f2',
			paddingBottom: 10
		},
		cardTitle: {
			fontWeight: 'bold'
		},
		label: {
			opacity: 0.5,
			marginBottom: 10
		},
		chipToday: {
			right: 10,
			top: 10,
			position: 'absolute',
			borderWidth: 1,
			borderColor: theme.colors.primary,
			backgroundColor: theme.colors.primaryContainer,
			borderRadius: 50
		},
		chipTodayText: {
			color: theme.colors.primary
		},
		wrap: {
			marginTop: 10,
			flexDirection: 'row',
			flexWrap: 'wrap',
			gap: 10
		}
	})

export default DishCard
