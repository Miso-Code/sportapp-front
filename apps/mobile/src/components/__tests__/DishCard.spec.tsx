import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'

import DishCard from '../DishCard'

jest.mock('dayjs')

describe('DishCard', () => {
	let component: ReactTestRenderer
	beforeEach(() => {
		component = renderer.create(
			<DishCard
				title='title'
				weekday='monday'
				category='category'
				calories={100}
				protein={100}
				carbs={100}
				fats={100}
			/>
		)
	})

	afterEach(() => {
		component.unmount()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})
})
