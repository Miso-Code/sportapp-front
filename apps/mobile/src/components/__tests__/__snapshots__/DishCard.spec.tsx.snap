// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DishCard should render correctly 1`] = `
<View
  collapsable={false}
  style={
    {
      "backgroundColor": "#f2f2f2",
      "borderRadius": 12,
      "shadowColor": "#000",
      "shadowOffset": {
        "height": 1,
        "width": 0,
      },
      "shadowOpacity": 0.15,
      "shadowRadius": 3,
    }
  }
  testID="card-container-outer-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "#f2f2f2",
        "borderRadius": 12,
        "elevation": 1,
        "flex": undefined,
        "paddingBottom": 10,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 1,
          "width": 0,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    testID="card-container"
  >
    <View
      style={
        [
          {
            "flexShrink": 1,
          },
          undefined,
        ]
      }
      testID="card"
    >
      <View
        borderRadiusStyles={{}}
        style={
          [
            {
              "paddingHorizontal": 16,
            },
            {
              "paddingVertical": 16,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "letterSpacing": 0.1,
                  "lineHeight": 20,
                },
                {
                  "marginBottom": 10,
                  "opacity": 0.5,
                },
              ],
            ]
          }
        >
          nutritionalPlan.weekday.monday
        </Text>
        <Text
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                },
                {
                  "fontWeight": "bold",
                },
              ],
            ]
          }
        >
          title
        </Text>
        <View
          style={
            {
              "flexDirection": "row",
              "flexWrap": "wrap",
              "gap": 10,
              "marginTop": 10,
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="chip-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderColor": "rgba(73, 69, 79, 0.12)",
                  "borderRadius": 8,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": undefined,
                  "flexDirection": "column",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="chip-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": false,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "borderRadius": 8,
                      },
                      {
                        "width": "100%",
                      },
                    ],
                  ]
                }
                testID="chip"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingLeft": 4,
                        "position": "relative",
                      },
                      {
                        "paddingLeft": 0,
                      },
                      {
                        "paddingRight": 0,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginVertical": 6,
                              "textAlignVertical": "center",
                            },
                            {
                              "color": "rgba(28, 27, 31, 0.38)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "marginLeft": 12,
                              "marginRight": 12,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    100
                     
                    nutritionalPlan.macros.calories
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="chip-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderColor": "rgba(73, 69, 79, 0.12)",
                  "borderRadius": 8,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": undefined,
                  "flexDirection": "column",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="chip-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": false,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "borderRadius": 8,
                      },
                      {
                        "width": "100%",
                      },
                    ],
                  ]
                }
                testID="chip"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingLeft": 4,
                        "position": "relative",
                      },
                      {
                        "paddingLeft": 0,
                      },
                      {
                        "paddingRight": 0,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginVertical": 6,
                              "textAlignVertical": "center",
                            },
                            {
                              "color": "rgba(28, 27, 31, 0.38)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "marginLeft": 12,
                              "marginRight": 12,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    100
                    g 
                    nutritionalPlan.macros.of_carbs
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="chip-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderColor": "rgba(73, 69, 79, 0.12)",
                  "borderRadius": 8,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": undefined,
                  "flexDirection": "column",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="chip-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": false,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "borderRadius": 8,
                      },
                      {
                        "width": "100%",
                      },
                    ],
                  ]
                }
                testID="chip"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingLeft": 4,
                        "position": "relative",
                      },
                      {
                        "paddingLeft": 0,
                      },
                      {
                        "paddingRight": 0,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginVertical": 6,
                              "textAlignVertical": "center",
                            },
                            {
                              "color": "rgba(28, 27, 31, 0.38)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "marginLeft": 12,
                              "marginRight": 12,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    100
                    g 
                    nutritionalPlan.macros.of_proteins
                  </Text>
                </View>
              </View>
            </View>
          </View>
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "transparent",
                "borderRadius": 8,
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="chip-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                {
                  "backgroundColor": "transparent",
                  "borderColor": "rgba(73, 69, 79, 0.12)",
                  "borderRadius": 8,
                  "borderStyle": "solid",
                  "borderWidth": 1,
                  "flex": undefined,
                  "flexDirection": "column",
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="chip-container"
            >
              <View
                accessibilityRole="button"
                accessibilityState={
                  {
                    "busy": undefined,
                    "checked": undefined,
                    "disabled": true,
                    "expanded": undefined,
                    "selected": false,
                  }
                }
                accessibilityValue={
                  {
                    "max": undefined,
                    "min": undefined,
                    "now": undefined,
                    "text": undefined,
                  }
                }
                accessible={true}
                collapsable={false}
                focusable={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    [
                      {
                        "borderRadius": 8,
                      },
                      {
                        "width": "100%",
                      },
                    ],
                  ]
                }
                testID="chip"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "paddingLeft": 4,
                        "position": "relative",
                      },
                      {
                        "paddingLeft": 0,
                      },
                      {
                        "paddingRight": 0,
                      },
                    ]
                  }
                >
                  <Text
                    numberOfLines={1}
                    selectable={false}
                    style={
                      [
                        {
                          "textAlign": "left",
                        },
                        {
                          "color": "rgba(28, 27, 31, 1)",
                          "writingDirection": "ltr",
                        },
                        [
                          {
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          [
                            {
                              "marginVertical": 6,
                              "textAlignVertical": "center",
                            },
                            {
                              "color": "rgba(28, 27, 31, 0.38)",
                              "fontFamily": "System",
                              "fontSize": 14,
                              "fontWeight": "500",
                              "letterSpacing": 0.1,
                              "lineHeight": 20,
                            },
                            {
                              "marginLeft": 12,
                              "marginRight": 12,
                            },
                            undefined,
                          ],
                        ],
                      ]
                    }
                  >
                    100
                    g 
                    nutritionalPlan.macros.of_fats
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "rgba(234, 221, 255, 1)",
              "borderRadius": 50,
              "position": "absolute",
              "right": 10,
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "top": 10,
            }
          }
          testID="chip-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              {
                "backgroundColor": "rgba(234, 221, 255, 1)",
                "borderColor": "rgba(103, 80, 164, 1)",
                "borderRadius": 50,
                "borderStyle": "solid",
                "borderWidth": 1,
                "flex": undefined,
                "flexDirection": "column",
                "shadowColor": "#000",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="chip-container"
          >
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": true,
                  "expanded": undefined,
                  "selected": false,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "overflow": "hidden",
                  },
                  [
                    {
                      "borderRadius": 50,
                    },
                    {
                      "width": "100%",
                    },
                  ],
                ]
              }
              testID="chip"
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "paddingLeft": 4,
                      "position": "relative",
                    },
                    {
                      "paddingLeft": 0,
                    },
                    {
                      "paddingRight": 0,
                    },
                  ]
                }
              >
                <Text
                  numberOfLines={1}
                  selectable={false}
                  style={
                    [
                      {
                        "textAlign": "left",
                      },
                      {
                        "color": "rgba(28, 27, 31, 1)",
                        "writingDirection": "ltr",
                      },
                      [
                        {
                          "fontFamily": "System",
                          "fontSize": 14,
                          "fontWeight": "500",
                          "letterSpacing": 0.1,
                          "lineHeight": 20,
                        },
                        [
                          {
                            "marginVertical": 6,
                            "textAlignVertical": "center",
                          },
                          {
                            "color": "rgba(29, 25, 43, 1)",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "500",
                            "letterSpacing": 0.1,
                            "lineHeight": 20,
                          },
                          {
                            "marginLeft": 12,
                            "marginRight": 12,
                          },
                          {
                            "color": "rgba(103, 80, 164, 1)",
                          },
                        ],
                      ],
                    ]
                  }
                >
                  nutritionalPlan.dishType.category
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
